src/mlpack/methods/lmnn/constraints.hpp

```diff
@@ -0,0 +1,188 @@
+                       const arma::Row<size_t>& labels,
+                       const size_t begin,
+                       const size_t batchSize);
+
+  /**
+   * Calculates k differently labeled nearest neighbors for each datapoint and
+   * writes them back to passed matrix.
+   *
+   * @param outputMatrix Coordinates matrix to store impostors.
+   * @param dataset Input dataset.
+   * @param labels Input dataset labels.
+   */
+  void Impostors(arma::Mat<size_t>& outputMatrix,
+                 const arma::mat& dataset,
+                 const arma::Row<size_t>& labels);
+
+  /**
+   * Calculates k differently labeled nearest neighbors & distances to
+   * impostors for each datapoint and writes them back to passed matrices.
+   *
+   * @param outputNeighbors Coordinates matrix to store impostors.
+   * @param outputDistance matrix to store distance.
+   * @param dataset Input dataset.
+   * @param labels Input dataset labels.
+   */
+  void Impostors(arma::Mat<size_t>& outputNeighbors,
+                 arma::mat& outputDistance,
+                 const arma::mat& dataset,
+                 const arma::Row<size_t>& labels);
+
+  /**
+   * Calculates k differently labeled nearest neighbors for a batch of dataset
+   * and writes them back to passed matrix.
+   *
+   * @param outputMatrix Coordinates matrix to store impostors.
+   * @param dataset Input dataset.
+   * @param labels Input dataset labels.
+   * @param begin Index of the initial point of dataset.
+   * @param batchSize Number of data points to use.
+   */
+  void Impostors(arma::Mat<size_t>& outputMatrix,
+                 const arma::mat& dataset,
+                 const arma::Row<size_t>& labels,
+                 const size_t begin,
+                 const size_t batchSize);
+
+  /**
+   * Calculates k differently labeled nearest neighbors & distances to
+   * impostors for a batch of dataset and writes them back to passed matrices.
+   *
+   * @param outputNeighbors Coordinates matrix to store impostors.
+   * @param outputDistance matrix to store distance.
+   * @param dataset Input dataset.
+   * @param labels Input dataset labels.
+   * @param begin Index of the initial point of dataset.
+   * @param batchSize Number of data points to use.
+   */
+  void Impostors(arma::Mat<size_t>& outputNeighbors,
+                 arma::mat& outputDistance,
+                 const arma::mat& dataset,
+                 const arma::Row<size_t>& labels,
+                 const size_t begin,
+                 const size_t batchSize);
+
+  /**
+   * Generate triplets {i, j, l} for each datapoint i and writes back generated
+   * triplets to matrix passed.
+   *
+   * @param outputMatrix Coordinates matrix to store triplets.
+   * @param dataset Input dataset.
+   * @param labels Input dataset labels.
+   */
+  void Triplets(arma::Mat<size_t>& outputMatrix,
+                const arma::mat& dataset,
+                const arma::Row<size_t>& labels);
+
+  //! Access the value of k.
```

Comment:
Do you think we could improve the comment here? e.g. ``Get the number of target neighbors (k).``.

Reply:
Done!